src/Compiler/Checking/NicePrint.fs

```diff
@@ -1572,7 +1572,13 @@ module TastDefinitionPrinting =
         let lhs = ConvertNameToDisplayLayout (tagRecordField >> mkNav fld.DefinitionRange >> wordL) fld.DisplayNameCore
         let lhs = (if isClassDecl then layoutAccessibility denv fld.Accessibility lhs else lhs)
         let lhs = if fld.IsMutable then wordL (tagKeyword "mutable") --- lhs else lhs
-        let fieldL = (lhs |> addColonL) --- layoutType denv fld.FormalType
+        let fieldL =
+            let rhs =
+                match fld.FormalType with
```

Comment:
Please never match on `TType` without first calling `stripTyEqns`

```fsharp
match stripTyEqns g fld.FormalType with 
| ...
```

Reply:
Actually in this case you want `stripTyparEqns`. This removes type inference equations, but not F# abbreviations e.g. consider

```fsharp
type IntFun = int -> int

type R = { Field: IntFun }
```
For anonymous records, type inference equations can be present I believe.  It's hard to construct test cases for these but it will be something like


```fsharp
let f x = 
    x 3 // force x to be solved to be a function
    {|  Field = x |}  // now use the type of x in an inference type
```

But anyway, don't worry bout the test case - just always remember not to match on TType directly without doing the appropariate strip.